core/src/main/java/org/hisp/dhis/android/core/event/search/EventQueryCollectionRepository.kt

```diff
@@ -0,0 +1,315 @@
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.android.core.event.search
+
+import androidx.lifecycle.LiveData
+import androidx.paging.DataSource
+import androidx.paging.PagedList
+import androidx.paging.PagingData
+import dagger.Reusable
+import io.reactivex.Single
+import kotlinx.coroutines.flow.Flow
+import org.hisp.dhis.android.core.arch.repositories.collection.ReadOnlyWithUidCollectionRepository
+import org.hisp.dhis.android.core.arch.repositories.filters.internal.EqFilterConnector
+import org.hisp.dhis.android.core.arch.repositories.filters.internal.EventDataFilterConnector
+import org.hisp.dhis.android.core.arch.repositories.filters.internal.ListFilterConnector
+import org.hisp.dhis.android.core.arch.repositories.filters.internal.PeriodFilterConnector
+import org.hisp.dhis.android.core.arch.repositories.filters.internal.ScopedFilterConnectorFactory
+import org.hisp.dhis.android.core.arch.repositories.`object`.ReadOnlyObjectRepository
+import org.hisp.dhis.android.core.arch.repositories.scope.RepositoryScope.OrderByDirection
+import org.hisp.dhis.android.core.common.AssignedUserMode
+import org.hisp.dhis.android.core.common.DateFilterPeriod
+import org.hisp.dhis.android.core.common.DateFilterPeriodHelper
+import org.hisp.dhis.android.core.common.State
+import org.hisp.dhis.android.core.event.Event
+import org.hisp.dhis.android.core.event.EventCollectionRepository
+import org.hisp.dhis.android.core.event.EventDataFilter
+import org.hisp.dhis.android.core.event.EventFilter
+import org.hisp.dhis.android.core.event.EventFilterCollectionRepository
+import org.hisp.dhis.android.core.event.EventObjectRepository
+import org.hisp.dhis.android.core.event.EventStatus
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnitMode
+import javax.inject.Inject
+
+@Reusable
+@Suppress("TooManyFunctions")
+class EventQueryCollectionRepository @Inject internal constructor(
+    private val eventCollectionRepositoryAdapter: EventCollectionRepositoryAdapter,
+    eventFilterRepository: EventFilterCollectionRepository,
+    scope: EventQueryRepositoryScope,
+) : ReadOnlyWithUidCollectionRepository<Event> {
+    private val connectorFactory: ScopedFilterConnectorFactory<
+        EventQueryCollectionRepository,
+        EventQueryRepositoryScope,
+        >
+    private val eventFilterRepository: EventFilterCollectionRepository
+
+    @JvmField
+    val scope: EventQueryRepositoryScope
+
+    init {
```

Comment:
I think this "init" block can be removed. Properties "eventFilterRepository" and "scope" can be used directly from the constructor by adding "private val". The "connectorFactory" can be initialized inline with the declaration